### PR TITLE
Remove noise padding payload (spec removed it)

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
Having some issues with NBC, don't merge yet I guess

File: libp2p/protocols/secure/noise.nim [hunk 1]
```diff
@@ -286,38 +286,23 @@ proc sendHSMessage(sconn: Connection; buf: seq[byte]) {.async.} =
...
 
   if result.len > uint16.high.int:
     raise newException(NoiseOversizedPayloadError, "Trying to send an unsupported oversized payload over Noise")
 
-  trace "packed noise payload", inSize = payload.len, outSize = result.len
+  trace "packed noise payload", size = payload.len
 
 proc unpackNoisePayload(payload: var seq[byte]) =
   let
     besize = payload[0..1]
```
Comment (arnetheduck) [inline libp2p/protocols/secure/noise.nim:299]: this will crash if there's not at least 2 bytes

Comment (sinkingsugar) [reply libp2p/protocols/secure/noise.nim:299]: nice catch, yeah I could not test properly so still the unpack is weird, I also could use an opinion on few lines after...

File: libp2p/protocols/secure/noise.nim [hunk 2]
```diff
@@ -286,38 +286,23 @@ proc sendHSMessage(sconn: Connection; buf: seq[byte]) {.async.} =
...
     size = uint16.fromBytesBE(besize).int
 
-  if size > (payload.len - 2):
+  if size != (payload.len - 2):
```
Comment (sinkingsugar) [inline libp2p/protocols/secure/noise.nim:302]: @arnetheduck does this change makes sense to you? 
I mean I guess that's exactly correct now... but invalidates any possible (wrong) padding.

File: libp2p/protocols/secure/noise.nim (hunk 1 above)
Comment (sinkingsugar) [reply libp2p/protocols/secure/noise.nim:299]: That would be a defect btw? I guess so

File: libp2p/protocols/secure/noise.nim (hunk 2 above)
Comment (arnetheduck) [reply libp2p/protocols/secure/noise.nim:302]: doesn't the standard say what to do with padded packets? 

If it doesn't, I'd generally lean towards allowing padding but not producing it, meaning using the original condition here.

I'd also generally avoid mutating the passed-in `var` most likely, instead trying to use the data in-place without copying it - for example turning this into a template and using `toOpenArray` etc.

Comment (arnetheduck) [reply libp2p/protocols/secure/noise.nim:302]: Oh, and I'd also include a link to the standard and explain the situation in a comment.

Comment (sinkingsugar) [reply libp2p/protocols/secure/noise.nim:302]: The spec does not say yeah, let's do like you said.

About the template, hmm what's the difference with `var` mutation? there should be no copy, but I will check again call-sites, indeed `toOpenArray` now is definitely an option too.

**Pull request merged**